lib/lrama/reporter.rb

```diff
@@ -0,0 +1,30 @@
+# rbs_inline: enabled
+# frozen_string_literal: true
+
+module Lrama
+  class Reporter
+    include Lrama::Trace::Duration
+
+    # @rbs (**Hash[Symbol, bool] options) -> void
+    def initialize(**options)
+      @options = options
+      @rules = Lrama::Report::Rules.new(**options)
+      @terms = Lrama::Report::Terms.new(**options)
+      @conflicts = Lrama::Report::Conflicts.new
+      @grammar = Lrama::Report::Grammar.new(**options)
+      @states = Lrama::Report::States.new(**options)
+    end
+
+    # @rbs (Lrama::States states, File io) -> void
+    def report(states, io)
+      report_duration(:report) do
+        logger = Lrama::Logger.new(io)
```

Comment:
Logger is an abstraction for managing which level of information is written to `io`. However `Reporter` is expected to always write the information. Please don't wrap `io` for report.

Reply:
Fix in 62ca9028be619e6b3773cd95dbbf823f3c10f38b